### DISCUSSION TITLE
Unable to use the Firely Terminal to resolve references during validation

### DISCUSSION THREAD

**Discussion by vshah11** [FirelyTeam/firely-net-sdk#2871] [General]
Hey Team,

I have been trying to use the Firely Terminal along with the hl7.fhir.r4.core specification to validate epic Fhir resources in my project. I keep getting the following error for any references present in the resource during the "fhir validate" command - 

```warning: Resolution of reference 'Patient/{id}' using the Resolver SDK failed: The method or operation is not implemented.```

Is this feature only available while using the dotnet sdk or am I using the terminal incorrectly. Any suggestions/help resolving this issue is greatly appreciated.

**Comment by mmsmits** [comment]
Hi @vshah11 ,
Fortunately, these are just warnings, not errors. It means that Firely Terminal hasn't implemented a way to resolve referenced resources for validation too. This is not a big problem if you just want to check the validation of the current resource.

Validation of the resource will never fail because of this. 

I will ask my Firely Terminal colleagues if they can give you more information about if you turn this feature on.

**Comment by vshah11** [comment]
Hey @mmsmits ,

Thank you for the reply. I understand that they are warning and not errors so it does not fail the validation for us. 

We have multiple resources being validated and there are some interesting references within them. The terminal currently gives the same warning message for each of them and we were just curious to understand if there is a way to correctly implement it. Any help from your Firely Terminal colleagues will be really appreciated.

**Comment by vshah11** [comment]
Hey @mmsmits ,

Could you find any of your Firely Terminal colleagues to help me on this?
